From e7d1e49f883326cc74f8e5a0b2a6b7da7ba8e309 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Thu, 11 Jun 2015 17:01:16 +0800
Subject: [PATCH 0604/5242] MLK-11093-2 dts: imx6ul-14x14-ddr3-arm2-emmc:
 enable HS200 support

commit  88a2f7f341e9c2421c61b92e1efd29e0674ca0f2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add ultra high speed mode pinctrl states to support eMMC HS200.
HW rework needed that changing both NVCC_SD and NVCC_NAND to 1.8v.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit f0f86b3bcd0ae1b1405f752862bde9745222372d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts
index 234d3ce..2e35ed6 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts
@@ -9,12 +9,15 @@
 #include "imx6ul-14x14-ddr3-arm2.dts"
 
 &usdhc1 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc1_8bit>;
+	pinctrl-1 = <&pinctrl_usdhc1_8bit_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_8bit_200mhz>;
 	bus-width = <8>;
 	cd-gpios = <>;
 	wp-gpios = <>;
 	vmmc-supply = <>;
+	tuning-step = <2>;
 	non-removable;
 	status = "okay";
 };
-- 
1.7.9.5

